artemis-server/src/main/java/org/apache/activemq/artemis/core/server/cluster/BackupManager.java

```diff
@@ -267,7 +268,8 @@ public void run() {
                   ClientSessionFactoryInternal backupSessionFactory = localBackupLocator.connect();
```

Comment:
Can this call also block?

Reply:
stopping the executor really should be bound, first stop and wait for completion, then whack it via shutdown now. I guess with agressive default tolerances.
keeping track of outstanding blocking commands as they are executed will be tedious :-)

Reply:
> Can this call also block?

It has the logic to retry connecting in a loop but I'm now aware if is happening: i suppose yes given that the old code was tracking it with `announcingBackup = true;`

> keeping track of outstanding blocking commands as they are executed will be tedious :-)
That's correct, the problem here is who has ownership on `ClusterControl`: if is `BackupManager` it should be in charge of closing a leaking one and is up to others (an existing connection), to unblock any blocked call.